### ISSUE TITLE
a11y: revision button lacks sufficient context

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#78138]
### Description

In the post sidebar, the revision row includes a number on the button, clicking which switches to the revision diff editor.

<img width="321" height="160" alt="Image" src="https://github.com/user-attachments/assets/5ce7f9e0-965e-43c8-9503-f4ac95d14f86" />

Sighted users can understand that this is a revision number because the text "Revisions" is on the left side. However, for users who rely on screen readers, when they focus on the button, it is only announced as `{number} button`, making it unclear what this button is for.

This issue has likely existed for some time, but since a new visual revision diff will be introduced in 7.0, it might be worth fixing in the 7.0 release.

This issue should be resolvable by adding an aria-label in addition to the visual text. Similar efforts are underway in other panels.

Perhaps a label like `Open revisions screen: {revision_number} revisions` would be appropriate.

### Step-by-step reproduction instructions

- Launch your screen reader.
- Open the post sidebar.
- Focus the revision button

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure
